mono/mini/interp/interp.c

```diff
@@ -479,8 +480,9 @@ stackval_from_data (MonoType *type, stackval *result, char *data, gboolean pinvo
 }
 
 static void inline
-stackval_to_data (MonoType *type, stackval *val, char *data, gboolean pinvoke)
+stackval_to_data (MonoType *type_, stackval *val, char *data, gboolean pinvoke)
 {
+	MonoType *type = mini_native_type_replace_type (type_);
```

Comment:
instead of replacing it in the callee, wouldn't it be better to replace it in the caller?